cmd/rest-server/main.go

```diff
@@ -72,12 +76,47 @@ func tlsSettings() (bool, string, string, error) {
 	} else {
 		cert = filepath.Join(server.Path, "public_key")
 	}
-	return server.TLS, key, cert, nil
+	return server.TLS, key, cert, server.TLSClientCerts, nil
+}
+
+func addClientCerts(p *x509.CertPool, fp string) error {
+	data, err := ioutil.ReadFile(fp)
+	if err != nil {
+		return err
+	}
+
+	if !p.AppendCertsFromPEM(data) {
+		return errors.New("PEM file contains invalid certificate")
+	}
+
+	return nil
+}
+
+func listenAndServeTLS(addr, certFile, keyFile, clientCertsFile string, handler http.Handler) error {
+	httpServer := http.Server{Addr: addr, Handler: handler}
+	if clientCertsFile != "" {
+		clientCerts := x509.NewCertPool()
+		err := addClientCerts(clientCerts, clientCertsFile)
+		if err != nil {
+			return err
+		}
+
+		authType := tls.VerifyClientCertIfGiven
+		if server.NoHTTPAuth {
+			authType = tls.RequireAndVerifyClientCert
+		}
+		httpServer.TLSConfig = &tls.Config{
+			ClientCAs:  clientCerts,
```

Comment:
why are we siggesting that we add the actual client certs to the server truststore for client authentication? Shouldn't we be suggesting that they provide a CA cert here, which is responsible for signing the client certs?